### DISCUSSION TITLE
Use IEEE Address instead of Short Address with ZbBridge

### DISCUSSION THREAD

**Discussion by dungdao191299** [arendst/Tasmota#13245] [Q&A] [answered]
Hi everyone, I am using Sonoff ZbBridge for my zigbee sensor and switch, but ZbBridge publishes payload with Short Address.  Is there a way for ZbBridge publish with IEEE Address in payload?

**Comment by s-hadinger** [comment, marked as answer]
No because this is the way Zigbee protocol works. All communications are based on short address, the long address is used at pairing and optionally for routing. Tasmota tries to keep track of IEEE address but there's no guarantee it knows the IEEE address or it is correct. Use `ZbInfo <addr>` to retrieve known information about a device.

Of course, Tasmota could still add its known IEEE address to the payload, but it would make them even longer than they are today. Why do you want IEEE address instead?

**Reply by sfromis** [reply]
If you really want to use the IEEE address in payloads, you could use ZbName to set the friendly name, and use options to have the friendly name used instead of short address. While I also do not like using the short address, my preference is to assign meaningful stable names using ZbName and so on, but if you think that IEEE addresses are "friendly", that's up to you.

**Reply by dungdao191299** [reply]
because the short address is randomly assigned when the device first connects. That mean after sensor is reset and reconnect to system, the short address will be change. But my system must be work with static address for each sensor ( so i want have IEEE address in payload to define the sensor even if sensor is reset

**Reply by sfromis** [reply]
Yes, that is why I also do not like using short address. It works excellently to neither use short nor IEEE address, but assign ZbName.

**Reply by dungdao191299** [reply]
sometime  my device is factory reset and i must go to my system to reconfig my device with new short address. So i want to use IEEE address to define my devcie, if ZbBridge can do it, I don't have to reconfigure the device when it's factory reset

**Reply by s-hadinger** [reply]
That's precisely what Zbname are for. They survive a factory reset and change of short address

**Reply by dungdao191299** [reply]
i think Zbname  is combine with short address, i will try again

**Reply by sfromis** [reply]
You can also use IEEE address when setting ZbName, not needing to rely on the potentially different short address for a fresh config, like a new device.

**Reply by s-hadinger** [reply]
ZBname internally stores both short and long address. If any message arrives from an IEEE address with a new short address, Zbname is immediately updated accordingly.
